.github/workflows/tests.yml

```diff
@@ -1,166 +1,254 @@
+
+      - run: brew style --display-cop-names
+
+      - run: brew man --fail-if-changed
+
+      - run: brew typecheck
+
+      - name: Run vale for docs linting
+        run: |
+          brew install vale
+          vale docs/
+
+      - name: Lint Dockerfile
+        run: |
+          brew install hadolint
+          hadolint Dockerfile
+
+  tests-no-compat-online:
+    needs: syntax
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Run brew tests
+        run: |
+          # brew tests doesn't like world writable directories
+          sudo chmod -R g-w,o-w /home/linuxbrew/.linuxbrew/Homebrew
+
+          brew tests --no-compat --online
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  tests-generic-online:
+    needs: syntax
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Run brew tests
+        run: |
+          # brew tests doesn't like world writable directories
+          sudo chmod -R g-w,o-w /home/linuxbrew/.linuxbrew/Homebrew
+
+          brew tests --generic --online
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  tests-online:
+    needs: syntax
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Run brew tests
+        run: |
+          # brew tests doesn't like world writable directories
+          sudo chmod -R g-w,o-w /home/linuxbrew/.linuxbrew/Homebrew
+
+          brew tests --online
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+          # set variables for coverage reporting
+          HOMEBREW_CODECOV_TOKEN: 3ea0364c-80ce-47a3-9fba-93a940d4b5d7
+
+          # These cannot be queried at the macOS level on GitHub Actions.
+          HOMEBREW_LANGUAGES: en-GB
+
+  tests-coverage-online:
+    needs: syntax
+    runs-on: macos-11.0
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Set up Xcode
+        run: sudo xcode-select --switch /Applications/Xcode_12.2.app/Contents/Developer
+
+      - name: Install brew tests dependencies
+        run: |
+          brew install subversion
+          Library/Homebrew/shims/scm/svn --homebrew=print-path
+          which svn
+          which svnadmin
+
+      - run: brew tests --online --coverage
+        env:
+          HOMEBREW_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+          # set variables for coverage reporting
+          HOMEBREW_CODECOV_TOKEN: 3ea0364c-80ce-47a3-9fba-93a940d4b5d7
+
+          # These cannot be queried at the macOS level on GitHub Actions.
+          HOMEBREW_LANGUAGES: en-GB
```

Comment:
All these which only run `brew tests` can be started using a `matrix`, something like:

```yml
matrix:
  include:
    - os: ubuntu-latest
      test_flags: --online
    - os: macos-latest
      test_flags: --online --coverage
```